tasker/job_template/generate.py

```diff
@@ -0,0 +1,36 @@
+from tasker.models import db, User, JobTemplate, Task, TaskStatus
+import datetime
+
+def generate_tasks(template_id):
+    template = JobTemplate.query.get(template_id)
+    user = template.owner
+    rep = template.repetition
+    interval = template.interval
+    if interval == 2:
+        rep *= 7
+    if interval == 3:
+        rep *= 30
+    start = template.starting_date
+    timestamp = datetime.datetime.fromtimestamp(start)
+    end = timestamp.replace(timestamp.year + 1)
+    while timestamp <= end:
+        task = Task.create_task(template.name, template.description,TaskStatus.Pending, timestamp)
+        task.owner = user
+        task.job_template = template
+        db.session.add(task)
+        db.session.commit()
```

Comment:
I have a hunch that this `db.session.commit()` should be outside the loop at the end.

Reply:
That will keep all the inserts in a single transaction- success or failure altogether.

Reply:
@jcaxmacher  All updates integrated. The changes have been tested in shell context. Recheck and see if that looks right.